webapp/javascript/components/Toolbar.tsx

```diff
@@ -179,22 +193,88 @@ function FocusOnSubtree({
   );
 }
 
-function HighlightSearch({ onHighlightChange, showMode }) {
+function HighlightSearch({
+  onHighlightChange,
+  showMode,
+  viewSide,
+  isSearchLinked,
+  linkedSearchQuery,
+  toggleLinkedSearch,
+  resetLinkedSearchSide,
+}) {
+  const [text, setText] = useState('');
+  const searchInputType =
+    isSearchLinked === undefined ? 'standalone' : 'linked';
+
+  if (isSearchLinked === true) {
+    if (viewSide === resetLinkedSearchSide) {
+      if (text !== '') {
+        setText('');
+      }
+    } else if (text !== linkedSearchQuery) {
+      setText(linkedSearchQuery);
+    }
+  }
+
+  const testId = `flamegraph-search${viewSide ? `-${viewSide}` : ''}`;
+
+  const searchSizeCls = `${
+    showMode === 'small' ? `${styles['search-small']}` : ''
+  }`;
+
+  const inputTypeCls = `${
+    searchInputType === 'linked' ? `${styles['linked-search-input']}` : ''
+  }`;
+
+  const linkedSearchCls = `${
+    isSearchLinked === true ? `${styles.active}` : ''
+  }`;
+
   return (
-    <DebounceInput
-      data-testid="flamegraph-search"
-      className={`${styles.search} ${
-        showMode === 'small' ? styles['search-small'] : ''
-      }`}
-      type="search"
-      name="flamegraph-search"
-      placeholder="Search…"
-      minLength={2}
-      debounceTimeout={100}
-      onChange={(e) => {
-        onHighlightChange(e.target.value);
-      }}
-    />
+    <>
+      <DebounceInput
+        data-testid={`${testId}`}
+        className={`${styles.search} ${searchSizeCls} ${inputTypeCls} ${linkedSearchCls}`}
+        type="search"
+        name="flamegraph-search"
+        placeholder="Search…"
+        minLength={2}
+        value={isSearchLinked === true ? linkedSearchQuery : text}
+        debounceTimeout={100}
+        onChange={(e) => {
+          onHighlightChange(e.target.value);
+        }}
+      />
+      {searchInputType === 'linked' && (
+        <button
+          data-testid={`link-search-btn-${viewSide}`}
+          type="button"
+          title={`${
+            isSearchLinked === true ? 'Unsync search term' : 'Sync search term'
+          }`}
+          onClick={() => {
+            toggleLinkedSearch();
+          }}
+          className={`${styles['linked-search-btn']} ${
+            isSearchLinked === true ? styles.active : ''
+          }`}
+        >
+          <svg
```

Comment:
Isn' there a fontAwesome version of this? Example https://github.com/pyroscope-io/pyroscope/blob/main/stories/Button.stories.tsx#L26

Reply:
It looks like that icon is different from the link one. @Rperry2174 let me know if we need to use that one instead. https://pyroscope.io/storybook/?path=/story/components-button--icon-only

Reply:
done. Found one matching link icon (solid though) https://fontawesome.com/v5.15/icons?d=gallery&p=2&q=link

![image](https://user-images.githubusercontent.com/7106086/148375230-80e55338-f126-4efc-a17e-4ef112f0b056.png)

![image](https://user-images.githubusercontent.com/7106086/148375208-d50f5f3a-f804-4641-a94e-b14f7a94a1e6.png)